tlvccmd/sys_utils/getty.c

```diff
@@ -86,28 +90,16 @@ void consolemsg(const char *str, ...)
 }
 
 
-#ifndef SUPER_SMALL
-char	Host[256], *Date = 0, *Time = 0;
+#if !SUPER_SMALL
+char	Host[20], *Date = 0, *Time = 0;
 
 void host(void) {
-    char *ptr;
-    int fp = open(_PATH_HOSTNAME,O_RDONLY), sz;
-
-    if (fp) {
-	sz = read( fp, Host, 255);
-	if (sz >= 0)
-	    Host[sz] = '\0';
-	else
-	    *Host = '\0';
-	close(fp);
-    }
-    for (ptr = Host; isprint(*ptr); ptr++)
-	continue;
-    while (ptr >= &Host[1] && ptr[-1] == ' ')
-	ptr--;
-    *ptr = '\0';
-    if (!*Host)
-	strcpy( Host, "LocalHost" );
+    char *host = getenv("HOSTNAME");
+
+    if (host == NULL)
+	strcpy(Host, "LocalHost");
+    else
+	strcpy(Host, host);
```

Comment:
If you're trying for min executable size, replace line 100 strcpy with initialization of char Host[] with Localhost via declaration statement. 

Line 102 strcpy is dangerous as length is not checked. Recommend `strncpy(Host, host, sizeof(Host)-1)`.

Reply:
Thanks, this is good. And yes, definitely going for min size. Along with `init`, `getty` is the one program always running. A byte here, a byte there...